### DISCUSSION TITLE
Not required DecimalField in nested Serializer gives error on empty input in browsable API

### DISCUSSION THREAD

**Discussion by jgadelange** [encode/django-rest-framework#7746] [Potential Issue]
I have a nested serializer with a `DecimalField(required=False)`. Implemented something like stated below:

``` python
class SubSerializer(serializers.Serializer):
    # This one gives errors when leaving the form input in browsable api empty:
    sub_field = serializers.DecimalField(required=False)


class MainSerializer(serializers.Serializer):
    # This shows behaviour I expext:
    main_field = serializers.DecimalField(required=False)
    sub = SubSerializer()
```

When I now use the browsable API I get a form as expected (a main field, sub field and a header for the Sub serializer). When I leave both inputs empty, I will only get an error for `sub_field` saying: `A valid number is required.` not for `main_field`.

**Comment by lovelydinosaur** [comment]
Probably worth including a screenshot of the browsable API form behavior for tickets like this.

The browsable API isn't yet intended to support nested fields in forms, so you should be using the raw content input box, but let's take a quick look at this all the same - see if we can make any improvements.

**Comment by jgadelange** [comment]
Good point, here is a screenshot:
![django_rest_framework](https://cloud.githubusercontent.com/assets/432366/9132374/4f86861e-3cf3-11e5-8aaa-ca4d2622405d.png)

Doing some research lead me to believe the behaviour comes from `Field.get_value(dictionary)` and the  `if html.is_html_input(dictionary)` within that method.

**Comment by lovelydinosaur** [comment]
Thanks @jgadelange.
